### PR TITLE
Implement query endpoint

### LINKED ISSUE #124: Query endpoint using existing DB query

### PR DESCRIPTION
**Pull request by richardhuaaa**
This implements the query endpoint and tests. I haven't implemented support for queries from a vector clock cursor yet due to the changes in https://github.com/xmtp/proto/pull/201/files, but will implement support for it in the next PR.

Closes #124

### CODE REVIEW

**Comment by richardhuaaa** [comment]
* **#131** <a href="https://app.graphite.dev/github/pr/xmtp/xmtpd/131?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @richardhuaaa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: pkg/db/subscription.go [hunk 1]
```diff
@@ -79,7 +79,9 @@ func (s *DBSubscription[ValueType]) poll() {
 	// Repeatedly query page by page until no more results
 	for {
 		results, lastID, err := s.query(s.ctx, s.lastSeenID, s.options.NumRows)
-		if err != nil {
+		if s.ctx.Err() != nil {
+			break
+		} else if err != nil {
```
Comment (richardhuaaa) [inline pkg/db/subscription.go:84]: It's pretty benign, but whenever I look at logs for the tests, they always end with entries saying `Error querying for DB subscription`. I assume this is happening during shutdown. This change was meant to silence those logs, but it didn't silence them - I suspect we're shutting things down in the wrong order. I can clean up the logs at a future date

Comment (neekolas) [reply pkg/db/subscription.go:84]: I do have a PR waiting in the wings that should simplify our shutdown flow. Not sure if it fixes this particular issue, but we can check after.

**Comment by richardhuaaa** [comment]
I've incorporated the changes from https://github.com/xmtp/proto/pull/201/files into this PR, hence the large line count, which is mostly generated code.

I've left queries using a vector clock cursor unimplemented in this PR, as it's more involved - will implement that as a separate PR.

File: pkg/api/publishWorker.go [hunk 2]
```diff
@@ -134,7 +136,9 @@ func (p *PublishWorker) publishStagedEnvelope(stagedEnv queries.StagedOriginator
 
 	// Try to delete the row regardless of if the gateway envelope was inserted elsewhere
 	deleted, err := q.DeleteStagedOriginatorEnvelope(context.Background(), stagedEnv.ID)
```
Comment (neekolas) [inline pkg/api/publishWorker.go:138]: We can also pass in `p.ctx` here so that this will fail if the server is in the process of shutting down

Comment (richardhuaaa) [reply pkg/api/publishWorker.go]: Totally missed this one, good catch

**Review by neekolas — APPROVED** [review]

**Pull request merged**